daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -270,14 +277,17 @@ func (l *logStream) collectBatch() {
 // publishBatch calls PutLogEvents for a given set of InputLogEvents,
 // accounting for sequencing requirements (each request must reference the
 // sequence token returned by the previous request).
-func (l *logStream) publishBatch(events []*cloudwatchlogs.InputLogEvent) {
+func (l *logStream) publishBatch(events []wrappedEvent) {
 	if len(events) == 0 {
 		return
 	}
 
+	// events in a batch must be sorted by timestamp
+	// see http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
 	sort.Sort(byTimestamp(events))
+	cwEvents := getCWEvents(events)
```

Comment:
Perhaps I don't understand the problem, but wouldn't simply removing the `sort.Sort(byTimestamp(events))` solve the problem?

If I read the code correctly, the events are *added* in the correct order (in `collectBatch`), so they should arrive here in the correct order.

*However*, due to `sort.Sort(byTimestamp(events))`, the events possibly get reordered, and end up in the wrong order (if two events have exactly the same time stamp)

Could you try if simply removing that sort resolves the issue?